Apps/Patient/src/Models/PatientModelV2.cs

```diff
@@ -0,0 +1,85 @@
+﻿//-------------------------------------------------------------------------
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+
+namespace HealthGateway.Patient.Models
+{
+    using System;
+    using System.Text.Json.Serialization;
+    using HealthGateway.Common.Models;
+
+    /// <summary>
+    /// The patient data model.
+    /// </summary>
+    public class PatientModelV2
+    {
+        /// <summary>
+        /// Gets or sets the health directed identifier.
+        /// </summary>
+        [JsonPropertyName("hdid")]
+        public string HdId { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the patient's PHN.
+        /// </summary>
+        [JsonPropertyName("personalHealthNumber")]
+        public string PersonalHealthNumber { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the patient's common name.
+        /// </summary>
+        [JsonPropertyName("commonName")]
+        public Name? CommonName { get; set; } = new();
+
+        /// <summary>
+        /// Gets or sets the patient's legal name.
+        /// </summary>
+        [JsonPropertyName("legalName")]
+        public Name LegalName { get; set; } = null!;
+
+        /// <summary>
+        /// Gets the patient's preferred name.
+        /// </summary>
+        [JsonPropertyName("preferredName")]
+        public Name PreferredName => this.CommonName ?? this.LegalName;
```

Comment:
As per @sslaws, the LegalName is expected to be populated from the EMPI service. Thus, if CommonName and LegalName is not populated we raise an error in the `ClientRegistriesDelegate`. I personally don't like empty objects, they give the sense that they're valid, the explicit null will break early and require less validation code to ensure a value is actually set. So I don't want it nullable.

Reply:
I would prefer the system to break early and not hide an invalid empty name object.

The `null!` is used quite extensively (83 references in a search) throughout the project, so I'm not favoring this suggestion. 

Can you please provide more reasoning to make the change, I would appreciate the insight if I'm missing something.

Reply:
The code in the delegate may populate CommonName, LegalName, or both.  To use the null coalescing operator in the PreferredName getter to determine if a particular name has been set, the uninitialized values of those names must be null.  If null is a valid value for a property, the property should be declared nullable.

(Declaring a non-nullable property with an initial value of `null!` should only be done when a non-null value is definitively assigned later.)

Reply:
LegalName should always be populated  by the service call to EMPI and we should fail early if it is not set.  From a new syntax perspective LegalName is a required property and so we could do:
public required Name LegalName { get; xxx; } where xxx is likely init but could ge set

If its a minor update then lets use the new required modifier, otherwise please raise a tech debt item in our backlog to make these changes in the future.

Reply:
Morning @MikeLyttle 

as per @sslaws comment its expected that LegalName should be assigned by the ClientRegistriesDelegate. 

following your comment:
> (Declaring a non-nullable property with an initial value of null! should only be done when a non-null value is definitively assigned later.)

Would this not indicate that the `LegalName` can be declared as `null!` as per the context anything but definitely setting the `LegalName` would be constitute an irregular/erroneous response, but not in the way that should break the entire sight, thus the forgiving `PreferredName`. I haven't yet stated that my intention is always to use `PreferredName` when referencing the patient in code, with that if for anywhere it is null, I would prefer a null reference exception to be thrown as the patient has not been created correctly.

The use of a default empty `Name` object will allow code to execute without error such as report headers but the result will just be an empty string, and this adds the burden that the consuming code needs to validate further.

On a note of the Required suggestion @sslaws, this could be done, but as we create PatientModels throughout the code, I'm sure we will have lots of compiler warning of null initialized properties, if we lean towards this approach.

Reply:
As per our discussion, and comments on the Azure Task, this has been resolved.